### PR TITLE
NERSC Hackathon work

### PR DESCRIPTION
**Pull request by jaharris87**
* Mostly bug fixes
* Some updates to reflect changes in thornado/weaklib interface
* GNU Makefile updates to allow GPU build of thornado/weaklib w/o GPU Castro/AMReX

### CODE REVIEW

File: Source/driver/meth_params.F90 [hunk 1]
```diff
@@ -801,8 +886,6 @@ subroutine ca_set_castro_method_params() bind(C, name="ca_set_castro_method_para
        gravity_type_int = 1
     else if (gravity_type == "PoissonGrav") then
        gravity_type_int = 2
-    else if (gravity_type == "PrescribedGrav") then
-       gravity_type_int = 3
```
Comment (zingale) [inline Source/driver/meth_params.F90]: why is this removed?

Comment (jaharris87) [reply Source/driver/meth_params.F90]: I'm not sure---I had made a change to `parse_castro_params.py`, and re-ran it with the change. Perhaps I didn't provide correct arguments to the script?

Comment (maxpkatz) [reply Source/driver/meth_params.F90]: Re-run the mk_params.sh script which invokves parse_castro_params.py for you.

Comment (zingale) [reply Source/driver/meth_params.F90]: okay... maybe someone made the change directly to that file and not the template in `development`.  Let me take a look.

Comment (jaharris87) [reply Source/driver/meth_params.F90]: > Re-run the mk_params.sh script which invokves parse_castro_params.py for you.

That's what I did initially.

Comment (maxpkatz) [reply Source/driver/meth_params.F90]: I updated the meth_params.template very recently, so perhaps you just need to sync again with development.

**Comment by jaharris87** [comment]
Merged `development` and re-ran the `mk_params.sh` script. Does this look okay now?

**Comment by zingale** [comment]
now it says there are 209 files changed -- is that what you expect?
